plugins/orders.cpp

```diff
@@ -0,0 +1,811 @@
+                    }
+
+                    if (it2->inorganic_bearing != -1)
+                    {
+                        condition["bearing"] = df::inorganic_raw::find(it2->inorganic_bearing)->id;
+                    }
+
+                    if (!it2->reaction_class.empty())
+                    {
+                        condition["reaction_class"] = it2->reaction_class;
+                    }
+
+                    if (!it2->has_material_reaction_product.empty())
+                    {
+                        condition["reaction_product"] = it2->has_material_reaction_product;
+                    }
+
+                    if (it2->has_tool_use != tool_uses::NONE)
+                    {
+                        condition["tool"] = enum_item_key(it2->has_tool_use);
+                    }
+
+                    // TODO: anon_1, anon_2, anon_3
+
+                    conditions.append(condition);
+                }
+
+                order["item_conditions"] = conditions;
+            }
+
+            if (!it->order_conditions.empty())
+            {
+                Json::Value conditions(Json::arrayValue);
+
+                for (auto it2 : it->order_conditions)
+                {
+                    Json::Value condition(Json::objectValue);
+
+                    condition["order"] = it2->order_id;
+                    condition["condition"] = enum_item_key(it2->condition);
+
+                    // TODO: anon_1
+
+                    conditions.append(condition);
+                }
+
+                order["order_conditions"] = conditions;
+            }
+
+            // TODO: anon_1
+
+            orders.append(order);
+        }
+    }
+
+    Filesystem::mkdir("dfhack-config/orders");
+
+    std::ofstream file("dfhack-config/orders/" + name + ".json");
+
+    file << orders << std::endl;
+
+    return file.good() ? CR_OK : CR_FAILURE;
+}
+
+static command_result orders_import_command(color_ostream & out, const std::string & name)
+{
+    if (!is_safe_filename(out, name))
+    {
+        return CR_WRONG_USAGE;
+    }
+
+    Json::Value orders;
+
+    {
+        std::ifstream file("dfhack-config/orders/" + name + ".json");
+
+        if (!file.good())
+        {
+            out << COLOR_LIGHTRED << "Cannot find orders file." << std::endl;
+            return CR_FAILURE;
+        }
+
+        file >> orders;
+
+        if (!file.good())
+        {
+            out << COLOR_LIGHTRED << "Error reading orders file." << std::endl;
+            return CR_FAILURE;
+        }
+    }
+
+    if (orders.type() != Json::arrayValue)
+    {
+        out << COLOR_LIGHTRED << "Invalid orders file: expected array" << std::endl;
+        return CR_FAILURE;
+    }
+
+    CoreSuspender suspend;
+
+    std::map<int32_t, int32_t> id_mapping;
+    for (auto it : orders)
+    {
+        id_mapping[it["id"].asInt()] = world->manager_order_next_id;
+        world->manager_order_next_id++;
+    }
+
+    for (auto & it : orders)
```

Comment:
The problem is that "order" and "condition" were already used inside the loops, so I didn't want to start using synonyms or other confusing stuff.